src/jit/importer.cpp

```diff
@@ -12881,7 +12947,9 @@ MATH_MAYBE_CALL_NO_OVF:         ovfl = false;
 
             JITDUMP(" %08X", resolvedToken.token);
 
-            if (!opts.IsReadyToRun())
+            fNeedsRuntimeLookup = info.compCompHnd->computeNeedsRuntimeLookupAndHandleType(&resolvedToken, FALSE);
```

Comment:
I changed the architecture a little bit, as we discussed, and made the impReadyToRunHelperToTree API return null if a runtime lookup is needed

---
In reply to: [62793570](https://github.com/dotnet/coreclr/pull/4519#discussion_r62793570) [](ancestors = 62793570)